bus-mapping/src/evm/opcodes.rs

```diff
@@ -50,18 +52,30 @@ use stop::Stop;
 use swap::Swap;
 
 /// Generic opcode trait which defines the logic of the
-/// [`Operation`](crate::operation::Operation) that should be generated for an
-/// [`ExecStep`](crate::circuit_input_builder::ExecStep) depending of the
-/// [`OpcodeId`] it contains.
+/// [`Operation`](crate::operation::Operation) that should be generated for one
+/// or multiple [`ExecStep`](crate::circuit_input_builder::ExecStep) depending
+/// of the [`OpcodeId`] it contains.
 pub trait Opcode: Debug {
     /// Generate the associated [`MemoryOp`](crate::operation::MemoryOp)s,
     /// [`StackOp`](crate::operation::StackOp)s, and
     /// [`StorageOp`](crate::operation::StorageOp)s associated to the Opcode
     /// is implemented for.
     fn gen_associated_ops(
         state: &mut CircuitInputStateRef,
```

Comment:
first parameter `state` contains `step` member, could we reuse it instead of importing new one?